storagemarket/impl/clientstates/client_states.go

```diff
@@ -314,6 +279,17 @@ func FailDeal(ctx fsm.Context, environment ClientDealEnvironment, deal storagema
 	return ctx.Trigger(storagemarket.ClientEventFailed)
 }
 
+func releaseReservedFunds(ctx fsm.Context, environment ClientDealEnvironment, deal storagemarket.ClientDeal) {
+	if !deal.FundsReserved.Nil() && !deal.FundsReserved.IsZero() {
+		err := environment.Node().ReleaseFunds(ctx.Context(), deal.Proposal.Client, deal.FundsReserved)
```

Comment:
If you want to use continue using deal.FundsReserved, you probably need to to undelete line 62 above